packages/database/src/object-list.ts

```diff
@@ -1,56 +1,64 @@
-'use server';
-
-import {objectLists, insertObjectItemSchema, objectItems} from './db/schema';
+import {objectLists, objectItems, insertObjectItemSchema} from './db/schema';
+import {InferSelectModel} from 'drizzle-orm';
 import {db} from './db/connection';
 import {iriToHash} from './iri-to-hash';
+import {DBQueryConfig} from 'drizzle-orm';
 
-async function getListsByCommunityId(communityId: string) {
-  return db.query.objectLists.findMany({
-    where: (objectLists, {eq}) => eq(objectLists.communityId, communityId),
-  });
+interface Option {
+  withObjects?: boolean;
+  limitObjects?: number;
+}
+
+interface ObjectList extends InferSelectModel<typeof objectLists> {
+  objects?: InferSelectModel<typeof objectItems>[];
 }
 
-async function getCommunityListsWithObjects(communityId: string) {
+export async function getByCommunityId(
```

Comment:
Should we validate the incoming options?

Reply:
Yes. I even already have a schema for the validation. Validation added in commit: https://github.com/colonial-heritage/colonial-collections/pull/248/commits/034c0980a4d861a480983ff5da551fa53ff9c451